trigger.js

```diff
@@ -215,39 +224,76 @@ class TriggerHappy {
+        const tkw = token.width * canvas.scene.data.grid / 2;
+        const tkh = token.height * canvas.scene.data.grid / 2;
+
+        const motion = new Ray({x: token.x + tkw, y: token.y  + tkh}, {x: finalX + tkw, y: finalY  + tkh});
 
         // don't trigger on tokens that are already captured
-        targets = targets.filter(target => token.x + tw !== target.center.x || token.y + th !== target.center.y)
-        
+        targets = targets.filter(target =>  !this._tokenContains(target, {x: token.x + tkw, y: token.y  + tkh}));
+
         // sort list by distance from start token position
         targets.sort((a , b) => targets.sort((a, b) => Math.hypot(token.x - a.x, token.y - a.y) - Math.hypot(token.x - b.x, token.y - b.y)))
         
         for (let target of targets) {
-            // test motion vs token diagonals
-            if (motion.intersectSegment([target.x, target.y, target.x + target.w, target.y + target.h])
-            || motion.intersectSegment([target.x, target.y + target.h, target.x + target.w, target.y])) {
-                update.x = target.center.x - tw;
-                update.y = target.center.y - th;
-                return true;
+            const tx = target.x;
+            const ty = target.y;
+            const tw = target.w;
+            const th = target.h;
+            const tgw = target.data.width; // target token width in grid units
+            const tgh = target.data.height; // target token height in grid units
+                // test motion vs token diagonals
+            if (tgw > 1 && tgh > 1 && tgw * tgh > 4) {
+                // big token so do boundary lines
+                var intersects = ( motion.intersectSegment([tx,      ty,      tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx + tw, ty     ])
+                                || motion.intersectSegment([tx + tw, ty + th, tx,      ty + th])
+                                || motion.intersectSegment([tx,      ty,      tx,      ty + th]))
+            } else  // just check the diagonals
+                var intersects = (motion.intersectSegment([tx,       ty,      tx + tw, ty + th])
+                               || motion.intersectSegment([tx,       ty + th, tx + tw, ty     ]));
+            if (intersects) {
+                if (tgw === 1 && tgh === 1) { // simple case size 1 target, return straight away.
+                    update.x = target.center.x - tkw;
+                    update.y = target.center.y - tkh;
+                    return true;
+                }
+                
+                let corners = [];
+                const gridSize = canvas.grid.size;
+                // Create a grid of the squares covered by the target token
+                [...Array(tgw).keys()].forEach(xc => [...Array(tgh).keys()].forEach(yc => 
+                    corners.push({x: target.data.x + xc * gridSize, y: target.data.y + yc * gridSize})
+                ))

```

Comment:
This is... ugly unfortunately :s I'd much prefer a for loop inside another, as it would be much easier to read.

Reply:
If only javescript had a proper range(), but since it's not going to get merged it does not matter.

Reply:
can always be a one liner which is really easy to read
corners = Array(tgw).fill(Array(tgh).fill(0)).map((v,i) => v.map((_,j) => {return {x:target.data.x + i * gridSize, y: target.data.y + j * gridSize}}))